keyboards/aki27/cocot46plus/cocot46plus.c

```diff
@@ -0,0 +1,280 @@
+        }
+        if (v_rev != 0) {
+            if (mouse_report.v + v_rev > 127) {
+                v_rev = 127 - mouse_report.v;
+            } else if (mouse_report.v + v_rev < -127) {
+                v_rev = -127 - mouse_report.v;
+            }
+            mouse_report.v += v_rev;
+            v_acm -= v_rev << scrl_div_array[cocot_config.scrl_div];
+        }
+
+        mouse_report.x = 0;
+        mouse_report.y = 0;
+    } else {
+        mouse_report.x = x_rev;
+        mouse_report.y = y_rev;
+    }
+
+    return pointing_device_task_user(mouse_report);
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t* record) {
+    // xprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed);
+    
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+
+    if (keycode == CPI_SW && record->event.pressed) {
+        cocot_config.cpi_idx = (cocot_config.cpi_idx + 1) % CPI_OPTION_SIZE;
+        eeconfig_update_kb(cocot_config.raw);
+        pointing_device_set_cpi(cpi_array[cocot_config.cpi_idx]);
+    }
+
+    if (keycode == SCRL_SW && record->event.pressed) {
+        cocot_config.scrl_div = (cocot_config.scrl_div + 1) % SCRL_DIV_SIZE;
+        eeconfig_update_kb(cocot_config.raw);
+    }
+
+    if (keycode == ROT_R15 && record->event.pressed) {
+        cocot_config.rotation_angle = (cocot_config.rotation_angle + 1) % ANGLE_SIZE;
+        eeconfig_update_kb(cocot_config.raw);
+    }
+
+    if (keycode == ROT_L15 && record->event.pressed) {
+        cocot_config.rotation_angle = (ANGLE_SIZE + cocot_config.rotation_angle - 1) % ANGLE_SIZE;
+        eeconfig_update_kb(cocot_config.raw);
+    }
+
+    if (keycode == SCRL_IN && record->event.pressed) {
+        cocot_config.scrl_inv = -cocot_config.scrl_inv;
+        eeconfig_update_kb(cocot_config.raw);
+    }
+
+    if (keycode == SCRL_TO && record->event.pressed) {
+        {
+            cocot_config.scrl_mode ^= 1;
+        }
+    }
+
+    if (keycode == SCRL_MO) {
+        {
+            cocot_config.scrl_mode ^= 1;
+        }
+    }
+
+    return true;
+}
+
+void eeconfig_init_kb(void) {
+    cocot_config.cpi_idx = COCOT_CPI_DEFAULT;
+    cocot_config.scrl_div = COCOT_SCROLL_DIV_DEFAULT;
+    cocot_config.rotation_angle = COCOT_ROTATION_DEFAULT;
+    cocot_config.scrl_inv = COCOT_SCROLL_INV_DEFAULT;
+    cocot_config.scrl_mode = false;
+    eeconfig_update_kb(cocot_config.raw);
+    eeconfig_init_user();
+    adns5050_write_reg(0x22, 0b10000 | 0x80);
+}
+
+
+void matrix_init_kb(void) {
+    // is safe to just read CPI setting since matrix init
+    // comes before pointing device init.
+    cocot_config.raw = eeconfig_read_kb();
+    if (cocot_config.cpi_idx > CPI_OPTION_SIZE) // || cocot_config.scrl_div > SCRL_DIV_SIZE || cocot_config.rotation_angle > ANGLE_SIZE)
+    {
+        eeconfig_init_kb();
+    }
+    matrix_init_user();
+}
```

Comment:
```suggestion
void keyboard_post_init_kb(void) {
    // is safe to just read CPI setting since matrix init
    // comes before pointing device init.
    cocot_config.raw = eeconfig_read_kb();
    if (cocot_config.cpi_idx > CPI_OPTION_SIZE) // || cocot_config.scrl_div > SCRL_DIV_SIZE || cocot_config.rotation_angle > ANGLE_SIZE)
    {
        eeconfig_init_kb();
    }
    keyboard_post_init_user();
}
```